tfjs-backend-webgpu/src/to_pixels_webgpu.ts

```diff
@@ -0,0 +1,88 @@
+/**
+ * @license
+ * Copyright 2023 Google LLC.
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * =============================================================================
+ */
+
+import {DataType} from '@tensorflow/tfjs-core';
+
+import {getMainHeaderString as main, PixelsOpType, WebGPUProgram} from './webgpu_program';
+import {computeDispatch, flatDispatchLayout} from './webgpu_util';
+
+export class ToPixelsProgram implements WebGPUProgram {
+  variableNames = ['Image'];
+  outputShape: number[];
+  shaderKey: string;
+  dispatchLayout: {x: number[]};
+  dispatch: [number, number, number];
+  workgroupSize: [number, number, number] = [64, 1, 1];
+  workPerThread = 4;
+  type: DataType;
+  textureFormat: GPUTextureFormat;
+  pixelsOpType = PixelsOpType.TO_PIXELS;
+  size = true;
+
+  constructor(
+      outShape: number[], type: DataType, textureFormat: GPUTextureFormat) {
+    this.outputShape = outShape;
```

Comment:
We can pass an outShape without channel like `[height, width]`. Then we don't need to make `workPerThread` to `4` and calculate a `flatIndex `again in shader. And the shader will be like below:
```
${main('index')} {
  if (index < uniforms.size) {
    var rgba = vec4<f32>(0.0, 0.0, 0.0, 1.0);
    for (var d = 0; d < uniforms.numChannels; d = d + 1) {
      let value = f32(inBuf[index * uniforms.numChannels + d]);
      ${calculateResult}
    }
    let coords = getCoordsFromIndex(index);
    textureStore(outImage, vec2<i32>(coords.yx), rgba);
  }
}

Reply:
Done, thanks!